From d84abb894a3cb6d8fe4244a47d7d450920677b22 Mon Sep 17 00:00:00 2001
From: Michel Alexandre Salim <salimma@fedoraproject.org>
Date: Sat, 3 Sep 2011 12:26:56 +0200
Subject: [PATCH] Access Account fields directly, not through getter functions

Per David Zeuthen
(https://bugzilla.redhat.com/show_bug.cgi?id=735341#c2):

gdbus-codegen(1) now uses (skip) annotation on generated C property
getters - it is better, and thread-safe, to access the GObject property
directly via e.g. account_id

Signed-off-by: Michel Alexandre Salim <salimma@fedoraproject.org>
---
 src/sources.js |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/sources.js b/src/sources.js
index 629ff35..f703cd1 100644
--- a/src/sources.js
+++ b/src/sources.js
@@ -48,8 +48,8 @@ Source.prototype = {
             this.object = params.object;
 
             let account = this.object.get_account();
-            this.id = account.get_id();
-            this.name = account.get_provider_name();
+            this.id = account.id;
+            this.name = account.provider_name;
         } else {
             this.id = params.id;
             this.name = params.name;
-- 
1.7.6.1


